aiodocker/swarm.py

```diff
@@ -59,13 +58,57 @@ def __init__(self, docker):
 
         return response
 
+    async def join(
+            self,
+            *,
+            remote_addrs: List[str],
+            listen_addr: str='0.0.0.0:2377',
+            join_token: str,
+            advertise_addr: str=None,
+            data_path_addr: str=None
+            ) -> bool:
+        """
+        Join a swarm.
+
+        Args:
+            listen_addr
+                Used for inter-manager communication
+
+            advertise_addr
+                Externally reachable address advertised to other nodes.
+
+            data_path_addr
+                Address or interface to use for data path traffic.
+
+            remote_addrs
+                Addresses of manager nodes already participating in the swarm.
+
+            join_token
+                Secret token for joining this swarm.
+        """
+
+        data = {
+            "RemoteAddrs": remote_addrs,
```

Comment:
If we change the parameter type to `Iterable`, it would be safer to wrap this with `tuple()` or `list()`.